influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -727,6 +740,62 @@ pub(crate) struct TableBatchMap<'a> {
     pub(crate) table_batches: HashMap<String, TableBatch>,
 }
 
+/// The 32 byte SHA256 digest of the full tag set for a line of measurement data
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct SeriesId([u8; 32]);
+
+impl std::fmt::Display for SeriesId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", hex::encode(self.0))
+    }
+}
+
+/// Produce a [`SeriesId`] from a raw line of line protocol / parsed [`Series`]
+///
+/// If the tag set is sorted, we parse out the original tags from the raw line
+/// protocol, instead of formatting a new string.
+///
+/// If the tags are not sorted, then we need to sort them, and allocate a new
+/// string to format them in the original 'key=val,key=val` form.
+///
+/// Generally, InfluxDB clients should be sending tags sorted.
+fn hash_series_id(raw_line: &str, series: &mut Series) -> SeriesId {
+    let bytes = if let Some(tag_set) = &mut series.tag_set {
+        if is_sorted(tag_set) {
+            // extract the raw string; if we are here, then the following `expect`s should
+            // never panic, there should be a tag set in the raw line protocol
+            let tag_set_str = raw_line
+                .split_once(',')
+                .expect("raw line protocol contains comma separating measurement from tag set")
+                .1
+                .split_once(' ')
+                .expect("raw line protocol contains tag set followed by space")
+                .0;
+            Sha256::digest(tag_set_str)
+        } else {
+            // sort the tag set and format as string, unstable sorting is fine because
+            // there should never be two tag keys of the same value
+            tag_set.sort_unstable_by(|(a, _), (b, _)| a.cmp(b));
+            let tag_set_str = tag_set
+                .iter()
+                .map(|(k, v)| format!("{k}={v}"))
+                .collect::<Vec<String>>()
+                .join(",");
+            Sha256::digest(tag_set_str)
+        }
+    } else {
+        // for an empty tag set, we still need a series ID:
+        Sha256::digest("")
```

Comment:
We should probably just impl Default for SeriesId, replace this with SeriesId::default(), and that method is just the precomputed hash of `Sha256::digest("")` to shave off some cpu cycles if a tag set isn't specified often.

Reply:
Nice, I like that, will give it a shot.